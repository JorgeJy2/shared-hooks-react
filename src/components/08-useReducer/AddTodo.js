import React from 'react'
import { useForm } from '../02-useEffect/hooks/useForm';
import PropTypes from 'prop-types';

export const AddTodo = React.memo(({ handlerAdd }) => {
    console.log('CREATE ADD TODO :( ');
    const [{ description }, handlerInputChange, resetForm] = useForm({
        description: ''
    });

    const handlearSubtmit = ($event) => {
        $event.preventDefault();
        if (description.trim().length > 0) {
            handlerAdd({
                id: new Date().getTime(),
                describe: description,
                done: false
            });
            resetForm();
        }
    };

    return (
        <form className="form-add" onSubmit={handlearSubtmit}>
            <p>Add todo</p>
            <hr />
            <input
                className="form-control"
                type="text"
                placeholder="Todo description"
                name="description"
                value={description}
                onChange={handlerInputChange}
            />
            <button
                className="btn btn-primary mt-5"
                type="submit">
                Add todo
                </button>
        </form>
    )
}
)


AddTodo.propTypes = {
    handlerAdd: PropTypes.func.isRequired
};