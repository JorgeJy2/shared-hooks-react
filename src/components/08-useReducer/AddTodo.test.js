import React from 'react';
import { shallow } from 'enzyme';
import { AddTodo } from './AddTodo';

describe('Test in <TodoAdd />', () => {

    const handleAddTodo = jest.fn();

    const wrapper = shallow(
        <AddTodo handlerAdd={handleAddTodo}
        />
    )


    test('should snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('should no call fun  handleAddTodo', () => {

        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault() { } });

        expect(handleAddTodo).toHaveBeenCalledTimes(0);


    });

    test('should call fun handleAddTodo', () => {

        const value = 'Aprender Firestore';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault() { } });

        expect(handleAddTodo).toHaveBeenCalledTimes(1);
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object)); // { }
        expect(handleAddTodo).toHaveBeenCalledWith({
            id: expect.any(Number),
            describe: value,
            done: false
        });
        expect(wrapper.find('input').prop('value')).toBe('');
    });
})