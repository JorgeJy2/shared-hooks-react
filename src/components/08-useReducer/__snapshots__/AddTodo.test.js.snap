// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <TodoAdd /> should snapshot 1`] = `
<form
  className="form-add"
  onSubmit={[Function]}
>
  <p>
    Add todo
  </p>
  <hr />
  <input
    className="form-control"
    name="description"
    onChange={[Function]}
    placeholder="Todo description"
    type="text"
    value=""
  />
  <button
    className="btn btn-primary mt-5"
    type="submit"
  >
    Add todo
  </button>
</form>
`;
